src/org/etools/j1939_84/bus/j1939/packets/GenericPacket.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) 2020. Equipment & Tool Institute
+ */
+package org.etools.j1939_84.bus.j1939.packets;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import org.etools.j1939_84.bus.Packet;
+import org.etools.j1939_84.bus.j1939.packets.model.PgnDefinition;
+import org.etools.j1939_84.bus.j1939.packets.model.Spn;
+import org.etools.j1939_84.bus.j1939.packets.model.SpnDataParser;
+import org.etools.j1939_84.bus.j1939.packets.model.SpnDefinition;
+
+import static org.etools.j1939_84.J1939_84.NL;
+
+public class GenericPacket extends ParsedPacket {
+
+    private List<Spn> spns;
+    private PgnDefinition pgnDefinition;
+    private final SpnDataParser parser;
+
+    public GenericPacket(Packet packet) {
+        this(packet, new SpnDataParser());
+    }
+
+    GenericPacket(Packet packet, SpnDataParser parser) {
+        super(packet);
+        this.parser = parser;
+    }
+
+    public String getAcronym() {
+        return getPgnDefinition().acronym;
+    }
+
+    @Override
+    public String getName() {
+        return getPgnDefinition().label;
+    }
+
+    public List<Spn> getSpns() {
+        if (spns == null) {
+            spns = new ArrayList<>();
+
+            //Sort so these are in the order they are on the packet.
+            List<SpnDefinition> spnDefinitions = getPgnDefinition().spnDefinitions;
+            spnDefinitions.sort(Comparator.comparing(d -> (d.startByte << 8) + d.startBit));
```

Comment:
Why not startByte * 8 + startBit?

Reply:
Because this made more sense to my brain. :)